snapcraft/commands/remote.py

```diff
@@ -193,14 +198,25 @@ def _run_new_or_fallback_remote_build(self, base: str) -> None:
             run_legacy()
             return
 
-        if is_repo(Path().absolute()):
+        git_type = get_git_repo_type(Path().absolute())
+
+        if git_type == GitType.NORMAL:
             emit.debug(
                 "Running new remote-build because project is in a git repository"
             )
             self._run_new_remote_build()
             return
 
-        emit.debug("Running fallback remote-build")
+        if git_type == GitType.SHALLOW:
+            emit.debug("Current git repository is shallow cloned.")
+            emit.progress(
+                "Remote build for shallow clones is deprecated "
+                "and will be removed in core24",
+                permanent=True,
+            )
+            # fall-through to legacy remote-build
+
+        emit.progress("Running fallback remote-build", permanent=True)
```

Comment:
To align with the 4 other `emit.debug("Running [sic] remote-build because...` calls above, which we decided to keep out of brief mode.

```suggestion
        emit.debug("Running fallback remote-build")
```

I think the deprecation notice is fine.

Reply:
OK, but I really think we should notice user they are fallback to legacy mode since the output is significant difference